src/routes/relay/relay.legacy.controller.spec.ts

```diff
@@ -0,0 +1,87 @@
+import { CacheModule } from '@/datasources/cache/cache.module';
+import { TestCacheModule } from '@/datasources/cache/__tests__/test.cache.module';
+import configuration from '@/config/entities/__tests__/configuration';
+import { RequestScopedLoggingModule } from '@/logging/logging.module';
+import { TestLoggingModule } from '@/logging/__tests__/test.logging.module';
+import { NetworkModule } from '@/datasources/network/network.module';
+import { TestNetworkModule } from '@/datasources/network/__tests__/test.network.module';
+import { TestAppProvider } from '@/__tests__/test-app.provider';
+import { AccountDataSourceModule } from '@/datasources/account/account.datasource.module';
+import { TestAccountDataSourceModule } from '@/datasources/account/__tests__/test.account.datasource.module';
+import { INestApplication } from '@nestjs/common';
+import { faker } from '@faker-js/faker';
+
+describe('Relay controller', () => {
+  let app: INestApplication;
+  const supportedChainIds = Object.keys(configuration().relay.apiKey);
+
+  beforeEach(async () => {
+    jest.resetAllMocks();
+
+    const defaultConfiguration = configuration();
+    const testConfiguration = (): typeof defaultConfiguration => ({
+      ...defaultConfiguration,
+      features: {
+        ...defaultConfiguration.features,
+        relay: true,
+      },
+    });
+
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule.register(testConfiguration)],
+    })
+      .overrideModule(AccountDataSourceModule)
+      .useModule(TestAccountDataSourceModule)
+      .overrideModule(CacheModule)
+      .useModule(TestCacheModule)
+      .overrideModule(RequestScopedLoggingModule)
+      .useModule(TestLoggingModule)
+      .overrideModule(NetworkModule)
+      .useModule(TestNetworkModule)
+      .compile();
+
+    app = await new TestAppProvider().provide(moduleFixture);
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  describe('POST /v1/chains/:chainId/relay', () => {
+    it('should return 302 and redirect to the new endpoint', async () => {
+      const chainId = faker.helpers.arrayElement(supportedChainIds);
+      const safeAddress = faker.finance.ethereumAddress();
+      const data = faker.string.hexadecimal();
+
+      await request(app.getHttpServer())
+        .post(`/v1/relay/${chainId}`)
+        .send({
+          to: safeAddress,
+          data,
+        })
+        .expect(302)
+        .expect((res) => {
+          expect(res.get('location')).toBe('/v1/chains/:chainId/relay');
+        });
+    });
+  });
+  describe('GET /v1/relay/:chainId/:safeAddress', () => {
+    it('should return 302 and redirect to the new endpoint', async () => {
+      const chainId = faker.string.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+
+      await request(app.getHttpServer())
+        .get(`/v1/relay/${chainId}/${safeAddress}`)
+        .expect(302)
```

Comment:
Given my comment above we can use [301 Moved Permanently](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301) given that we are dealing with a `GET` request.

Reply:
Changed in 3f2db35fc53e88dc0419f8b40fba9a327e61c49b.